frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +166,896 @@ def qfunc(data, x, weights):
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit(np.pi / 4, np.pi / 3, np.pi / 2)
+    observed = compiled(np.pi / 4, np.pi / 3, np.pi / 2)
+
+    assert expected.get_nodes() == observed.get_nodes()
+
+
+def test_commute_controlled(backend):
+    """Test commute_controlled"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @partial(qml.transforms.commute_controlled, direction="right")
+        @qml.qnode(qml.device(device_name, wires=3), interface="jax")
+        def qfunc(theta):
+            qml.CZ(wires=[0, 2])
+            qml.X(2)
+            qml.S(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.Y(1)
+            qml.CRY(theta, wires=[0, 1])
+            qml.PhaseShift(theta / 2, wires=0)
+            qml.Toffoli(wires=[0, 1, 2])
+            qml.T(wires=0)
+            qml.RZ(theta / 2, wires=1)
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit(0.5)
+    observed = compiled(0.5)
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+def test_convert_to_numpy_parameters(backend):
+    """Test convert_to_numpy_parameters"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.convert_to_numpy_parameters
+        @qml.qnode(qml.device(device_name, wires=1), interface="jax")
         def qfunc():
-            """Example taken from PL tests"""
-            qml.Hadamard(1)
-            qml.Hadamard(0)
-            qml.PauliZ(0)
-            qml.Hadamard(3)
-            qml.Hadamard(5)
-            qml.T(5)
-            return (
-                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
-                qml.expval(qml.PauliX(0)),
-                qml.expval(qml.PauliZ(1)),
-                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
-                qml.expval(qml.PauliX(3)),
-                qml.expval(qml.PauliY(5)),
-            )
+            qml.S(0)
+            qml.RX(0.1234, 0)
+            return qml.expval(qml.X(0))
 
         return qfunc
 
     qnode_control = qnode_builder("default.qubit")
     qnode_backend = qnode_builder(backend)
-    expected = jax.jit(qnode_control)()
-    observed = qjit(qnode_backend)()
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="cond does not work with qml.RY")
```

Comment:
Is this a pennylane error or qjit error?

Reply:
Updated the reason to reflect that it is a qjit error.